contentcuration/contentcuration/frontend/channelEdit/components/edit/CategoryOptions.vue

```diff
@@ -0,0 +1,262 @@
+  }
+
+  const flattenCategories = obj => {
+    Object.keys(obj).forEach(key => {
+      if (key !== 'value' && key !== 'nested') {
+        dropdown.push({
+          text: key,
+          value: obj[key]['value'],
+        });
+      }
+      if (typeof obj[key] === 'object') {
+        flattenCategories(obj[key]);
+      }
+    });
+    return dropdown;
+  };
+
+  flattenCategories(categoriesObj);
+
+  export default {
+    name: 'CategoryOptions',
+    mixins: [constantsTranslationMixin, metadataTranslationMixin],
+    props: {
+      value: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data() {
+      return {
+        categoryText: null,
+        chipsSelected: [],
+        nested: true,
+      };
+    },
+    computed: {
+      categoriesList() {
+        return dropdown.map(category => {
+          return {
+            ...category,
+            text: this.translateMetadataString(camelCase(category.text)),
+            level: this.findDepth(Categories[category.text]),
+          };
+        });
+      },
+      dropdownSelected() {
+        return [
+          ...new Set(
+            this.chipsSelected.reduce((a, chip) => a.concat(this.findFamilyTreeIds(chip)), [])
+          ),
+        ];
+      },
+      selected: {
+        get() {
+          return this.value;
+        },
+        set(value) {
+          let items = [...value];
+          let currentItem = items[items.length - 1] || '';
+
+          this.handleCheckboxItem(currentItem, items);
+          this.$emit('input', this.chipsSelected);
+        },
+      },
+      dropdownSelectedObj() {
+        return this.dropdownSelected.map(item => {
+          return {
+            text: this.translateMetadataString(camelCase(CategoriesLookup[item])),
+            value: item,
+            level: this.findDepth(item),
+          };
+        });
+      },
+    },
+    watch: {
+      categoryText(val) {
+        return !val ? (this.nested = true) : (this.nested = false);
+      },
+    },
+    methods: {
+      treeItemStyle(item) {
+        return this.nested ? { paddingLeft: `${item.level * 24}px` } : {};
+      },
+      remove(item) {
+        this.chipsSelected.splice(this.chipsSelected.indexOf(item), 1);
+        this.$emit('input', this.chipsSelected);
+      },
+      tooltipHelper(id) {
+        return this.displayFamilyTree(dropdown, id)
+          .map(node => this.translateMetadataString(camelCase(node.text)))
+          .join(' - ');
+      },
+      findDepth(val) {
+        return val.split('.').length - 1;
+      },
+      displayFamilyTree(nodes, id) {
+        return nodes.filter(node => this.findFamilyTreeIds(id).includes(node.value));
+      },
+      findFamilyTreeIds(value) {
+        const family = [];
+        let familyMember = value;
+        while (familyMember) {
+          family.push(familyMember);
+          familyMember = familyMember.includes('.')
+            ? familyMember.substring(0, familyMember.lastIndexOf('.'))
```

Comment:
Just want to be sure I can follow what's happening here.. My understanding reading through is that the array is created by cycling through the id and each time removing the last value. i.e. if the initial value is `x.y.z`, then `family.push(x.y.z)` then, `familyMember` becomes `x.y`, `family.push(x.y)`, `familyMember` becomes `x`, `family.push(x)`. Is that right? So then family becomes an array with a set order of `[x, x.y, x.y.z]`?

Reply:
Added docstring for `findFamilyTreeIds`